app/services/authorization/user_allowed_service.rb

```diff
@@ -102,18 +102,18 @@ def allowed_to_in_project?(action, project, _options = {})
   # See allowed_to? for the actions and valid options.
   def allowed_to_globally?(action, _options = {})
     # Inactive users are never authorized
-    return false unless is_active_user?
+    return false unless authorizable_user?
     # Admin users are always authorized
     return true if user.admin?
 
     has_authorized_role?(action)
   end
 
   ##
-  # Only active users may perform actions
+  # Only users that are not locked may be granted actions
   # with the exception of a temporary-granted system user
-  def is_active_user?
-    user.active? || user.is_a?(SystemUser)
+  def authorizable_user??
+    user.active_or_registered? || user.is_a?(SystemUser)
```

Comment:
@ulferts  I initially thought that I could move the `.active?` check in here, but we do not only use these checks for the current logged in user but also for invited users - for example to identify which users are allowed watchers of an item. Please have a closer look if this makes sense but I believe it does for our use cases.

Reply:
I'd have thought it ideal if we where to only have `active?` in here as the registered/invited users should also not be allowed to perform most of the actions. To the best of my knowledge other mechanisms like not being able to log in prevent it would have been an extra check. But I also do not see a way to only allow `active?` in here with the current structure of the code, so having this weakened check in here is better than not having it at all.